pandas/tests/indexes/test_index_new.py

```diff
@@ -138,6 +142,24 @@ def test_constructor_mixed_nat_objs_infers_object(self, swap_objs):
         tm.assert_index_equal(Index(data), expected)
         tm.assert_index_equal(Index(np.array(data, dtype=object)), expected)
 
+    def test_constructor_datetime_inference(self):
+        # Index.__new__ inference to datetime matches Series
+
+        # mixed str/datetime
+        data = ["2012-12-31", datetime(2013, 1, 1), datetime(2013, 1, 2)]
+        result = Index(data)
+        expected = DatetimeIndex([Timestamp(x) for x in data])
```

Comment:
I don't find this an improvement (IMO we shouldn't infer strings just because timestamps are present), so if we want to (eventually) achieve consistency, I would rather deprecate this in the Series constructor.